pkg/apb/rhcc_registry.go

```diff
@@ -63,19 +63,73 @@ func (r RHCCRegistry) LoadSpecs() ([]*Spec, int, error) {
 func (r RHCCRegistry) imageToSpec(image *Image) (*Spec, error) {
 	r.log.Debug("RHCCRegistry::imageToSpec")
 	_spec := &Spec{}
-	// Setting ID to foo because Dao expects non blank IDs
-	_spec.Id = "foo_id"
-	var name = strings.SplitAfter(image.Name, "/")
-	_spec.Name = name[len(name)-1]
-	_spec.Image = image.Name
-	_spec.Description = image.Description
+
+	req, err := http.NewRequest("GET", r.config.Url+"/v2/"+image.Name+"manifests/latest", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	type label struct {
+		Spec    string `json:"com.redhat.apb.spec"`
+		Version string `json:"com.redhat.apb.version"`
+	}
+
+	type config struct {
+		Label label `json:"Labels"`
+	}
+
+	hist := struct {
+		History []map[string]string `json:"history"`
+	}{}
+
+	conf := struct {
+		Config *config `json:"config"`
+	}{}
+
+	err = json.NewDecoder(resp.Body).Decode(&hist)
+	if err != nil {
+		r.log.Error("Error grabbing JSON body from response: %s", err)
+		return nil, err
+	}
+
+	/*	if hist.History == nil {
+		r.log.Error("V1 Schema Manifest does not exist in registry")
+		return nil, nil
+	}*/
+
+	err = json.Unmarshal([]byte(hist.History[0]["v1Compatibility"]), &conf)
+	if err != nil {
+		r.log.Error("Error unmarshalling intermediary JSON response")
+		return nil, err
+	}
+
+	r.log.Debug(r.config.Url)
+	encodedSpec := conf.Config.Label.Spec
+	decodedSpecYaml, _err := b64.StdEncoding.DecodeString(encodedSpec)
```

Comment:
I would not create a new error variable here, I think that it should be err, imo